metricq_tools/summary.py

```diff
@@ -0,0 +1,294 @@
+        )
+        click.echo()
+
+        self._print_histogram(values)
+
+        click.echo()
+        click.echo()
+
+    def print(self):
+        for metric in self._metrics:
+
+            if self.values[metric]:
+                click.echo()
+                click.echo(click.style(f"Stats of metric {metric!r}:", fg="green"))
+                click.echo()
+
+                if self.print_intervals and self.last_timestamp[metric]:
+                    self._print_intervals_histogram(self.intervals[metric])
+
+                if self.print_values:
+                    self._print_values_histogram(self.values[metric])
+            else:
+                click.echo()
+                click.echo(
+                    click.style(f"No Data for metric {metric!r} received!", fg="red")
+                )
+                click.echo()
+
+        if self.print_stats:
+            self._print_statistics()
+
+    def _print_statistics(self):
+        click.echo(
+            click.style(
+                "Statistics",
+                fg="yellow",
+            )
+        )
+        click.echo()
+
+        table = list[list]()
+        table.append(["Minimum"])
+        table.append(["Maximum"])
+        table.append(["Average"])
+        table.append(["Median"])
+        table.append(["Standard deviation"])
+        table.append(["Arithmetic mean"])
+        table.append(["Variance"])
+        table.append(["Count"])
+
+        for metric in self._metrics:
+            if self.values[metric]:
+                table[0].append(np.amin(self.values[metric]))
+                table[1].append(np.amax(self.values[metric]))
+                table[2].append(np.average(self.values[metric]))
+                table[3].append(np.median(self.values[metric]))
+                table[4].append(np.std(self.values[metric]))
+                table[5].append(np.mean(self.values[metric]))
+                table[6].append(np.var(self.values[metric]))
+                table[7].append(len(self.values[metric]))
+            else:
+                table[0].append("n/a")
+                table[1].append("n/a")
+                table[2].append("n/a")
+                table[3].append("n/a")
+                table[4].append("n/a")
+                table[5].append("n/a")
+                table[6].append("n/a")
+                table[7].append(len(self.values[metric]))
+
+        click.echo(tabulate(table, headers=self._metrics, tablefmt="fancy_grid"))
+
+        click.echo()
+        click.echo()
+
+
+async def run_cmd(command):
+    click.echo(f"running... {command!r}")
+
+    proc = await asyncio.create_subprocess_shell(
+        command, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
+    )
+
+    stdout, stderr = await proc.communicate()
+
+    if stdout:
+        click.echo(stdout.decode())
+    if stderr:
+        click.echo(click.style(stderr.decode(), fg="red"))
+    click.echo(f"{command!r} exited with {proc.returncode}")
```

Comment:
Two suggestions here:

1. Similarly to the above, this should probably log at level `INFO`, except when `proc.returncode != 0` (then it should log an `ERROR`).  Something like

```python
    exit_msg = f"{command!r} done (exited with {proc.returncode})"
    if proc.returncode:
        logger.error(exit_msg)
    else:
        logger.info(exit_msg)
```

2. I think we should return the exit code of the subprocess here, so that we can `sys.exit(...)` with the correct status code in the main function.